crates/iota-transaction-checks/src/lib.rs

```diff
@@ -190,6 +193,46 @@ mod checked {
         Ok(input_objects.into_checked())
     }
 
+    #[instrument(level = "trace", skip_all)]
+    pub fn check_move_authenticator_input(
+        protocol_config: &ProtocolConfig,
+        reference_gas_price: u64,
+        transaction: &TransactionData,
```

Comment:
Let's pass the single elements instead of the TransactionData, this is more clear for the caller:
```suggestion
        gas_owner: IotaAddress,
        gas_budget: u64,
        gas_price: u64,
        gas: &[ObjectRef],
```